src/FSharpLint.Core/Framework/Ast.fs

```diff
@@ -327,9 +322,22 @@ module Ast =
         | SynExpr.Upcast(expression, synType, _)
         | SynExpr.Downcast(expression, synType, _) ->
             addMany [Type synType; Expression expression]
-        | SynExpr.LetOrUse(_, _, bindings, expression, _, _) ->
+        // regular let or use
+        | SynExpr.LetOrUse(_, _, _, false, bindings, expression, _, _) ->
             add <| Expression expression
             List.revIter (Binding >> add) bindings
+        // let! or use!
+        | SynExpr.LetOrUse(_, _, _, true, bindings, leftHandSide, _, _) ->
+            match bindings with
```

Comment:
Change based on the FCS release notes at https://github.com/dotnet/fsharp/blob/c1b13e3d4542d1de7439b34de00384934365aeb9/docs/release-notes/.FSharp.Compiler.Service/10.0.100.md?plain=1#L63